### ISSUE TITLE
cmd/go: error message for incorrect downloaded zip truncates last character of prefix

### ISSUE THREAD

**Issue by shoenig** [golang/go#27478]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
$ go version
go version go1.11 linux/amd64

### What did you do?

Created an incorrectly structured .zip file served up via GOPROXY. Noticed the error message regarding the proper file prefix within the zip is truncating the last character of the expected prefix name.

```
minifiers/minifiers.go:27:2: unknown import path "github.com/tdewolff/minify": zip for github.com/tdewolff/minify@v2.3.5+incompatibl has unexpected file github.com/tdewolff/minify@v2.3.5/.gitignore
minifiers/minifiers.go:28:2: unknown import path "github.com/tdewolff/minify/css": zip for github.com/tdewolff/minify@v2.3.5+incompatibl has unexpected file github.com/tdewolff/minify@v2.3.5/.gitignore
minifiers/minifiers.go:29:2: unknown import path "github.com/tdewolff/minify/html": zip for github.com/tdewolff/minify@v2.3.5+incompatibl has unexpected file github.com/tdewolff/minify@v2.3.5/.gitignore
...
```
e.g. `zip for github.com/tdewolff/minify@v2.3.5+incompatibl has unexpected file` is missing the last `e` in `incompatible`

I don't know why the source code is explicitly chopping off the last character - maybe it was assumed to be cleaning up a trailing slash:
https://github.com/golang/go/blob/master/src/cmd/go/internal/modfetch/fetch.go#L126

**Comment by unicod3** [comment]
it seems that solved at #27279

**Comment by agnivade** [comment]
@unicod3 - Thanks for spotting it ! Indeed it looks fixed.
